clients/src/main/resources/common/message/ListConfigResourcesResponse.json

```diff
@@ -16,18 +16,22 @@
 {
   "apiKey": 74,
   "type": "response",
-  "name": "ListClientMetricsResourcesResponse",
-  "validVersions": "0",
+  "name": "ListConfigResourcesResponse",
+  // Version 0 is used as ListClientMetricsResourcesResponse which returns all client metrics resources.
+  // Version 1 adds ResourceType to ConfigResources (KIP-1142).
+  "validVersions": "0-1",
   "flexibleVersions": "0+",
   "fields": [
       { "name": "ThrottleTimeMs", "type": "int32", "versions": "0+",
         "about": "The duration in milliseconds for which the request was throttled due to a quota violation, or zero if the request did not violate any quota." },
       { "name": "ErrorCode", "type": "int16", "versions": "0+",
         "about": "The error code, or 0 if there was no error." },
-      { "name": "ClientMetricsResources", "type": "[]ClientMetricsResource", "versions": "0+",
-        "about": "Each client metrics resource in the response.", "fields": [
-        { "name": "Name", "type": "string", "versions": "0+",
-          "about": "The resource name." }
+      { "name": "ConfigResources", "type": "[]ConfigResource", "versions": "0+",
+        "about": "Each config resource in the response.", "fields": [
+        { "name": "ResourceName", "type": "string", "versions": "0+",
+          "about": "The resource name." },
+        { "name": "ResourceType", "type": "int8", "versions": "1+", "ignorable": true,
```

Comment:
Should this be `"default": "16"`? I'm still researching this, so that's an open question.

Reply:
Good catch! It's reasonable to set default value as 16. The reason is that `AbstractResponse` doesn't keep version data. When users use `ListConfigResourcesResponse#clientMetricsResources`, it cannot use version to determine whether the data is v0 or v1. If resource type has default value as 16, the `ListConfigResourcesResponse#clientMetricsResponses` can always use resource type filter client metrics.